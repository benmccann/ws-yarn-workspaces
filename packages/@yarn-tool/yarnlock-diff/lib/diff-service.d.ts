/// <reference types="node" />
import { Diff } from "deep-diff";
import { Option } from "fp-ts/lib/Option";
export declare function buildDiff(oldYarnLockContent: (Buffer | string)[], newYarnLockContent: (Buffer | string)[]): Option<Diff<{}, {}>[]>;
