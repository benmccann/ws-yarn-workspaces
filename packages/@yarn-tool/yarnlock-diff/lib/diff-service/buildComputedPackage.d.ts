/// <reference types="node" />
import { IComputedPackage } from './types';
import { ITSValueOrArray } from 'ts-type/lib/type/base';
export declare function buildComputedPackage(yarnLockContentList: ITSValueOrArray<Buffer | string>, alreadyComputedPackage?: IComputedPackage): IComputedPackage;
