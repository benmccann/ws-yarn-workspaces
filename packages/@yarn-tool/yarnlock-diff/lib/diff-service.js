"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildDiff = void 0;
const deep_diff_1 = require("@bluelovers/deep-diff");
const buildComputedPackage_1 = require("./diff-service/buildComputedPackage");
function buildDiff(oldYarnLockContent, newYarnLockContent) {
    var _a;
    const oldPacakges = buildComputedPackage_1.buildComputedPackage(oldYarnLockContent);
    const newPackages = buildComputedPackage_1.buildComputedPackage(newYarnLockContent);
    return (_a = deep_diff_1.diff(oldPacakges, newPackages)) !== null && _a !== void 0 ? _a : [];
}
exports.buildDiff = buildDiff;
//# sourceMappingURL=diff-service.js.map