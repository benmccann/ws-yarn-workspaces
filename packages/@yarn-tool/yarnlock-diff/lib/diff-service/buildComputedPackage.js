"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildComputedPackage = buildComputedPackage;
const yarnlock_parse_1 = require("@yarn-tool/yarnlock-parse");
const computeHashmapOfPackageAndVersionList_1 = require("./computeHashmapOfPackageAndVersionList");
function buildComputedPackage(yarnLockContentList, alreadyComputedPackage = {}) {
    if (!Array.isArray(yarnLockContentList)) {
        yarnLockContentList = [yarnLockContentList];
    }
    return yarnLockContentList
        .map(v => (0, yarnlock_parse_1.yarnLockParse)(v))
        .reduce(computeHashmapOfPackageAndVersionList_1.computeHashmapOfPackageAndVersionList, alreadyComputedPackage);
}
//# sourceMappingURL=buildComputedPackage.js.map