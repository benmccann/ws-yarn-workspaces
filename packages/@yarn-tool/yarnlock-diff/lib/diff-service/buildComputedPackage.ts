import { IComputedPackage } from './types';
import { yarnLockParse } from '@yarn-tool/yarnlock-parse/index';
import { computeHashmapOfPackageAndVersionList } from './computeHashmapOfPackageAndVersionList';
import { ITSValueOrArray } from 'ts-type/lib/type/base';

export function buildComputedPackage(yarnLockContentList: ITSValueOrArray<Buffer | string>, alreadyComputedPackage: IComputedPackage = {})
{
	if (!Array.isArray(yarnLockContentList))
	{
		yarnLockContentList = [yarnLockContentList];
	}

	return yarnLockContentList
		.map(v => yarnLockParse(v))
	.reduce(computeHashmapOfPackageAndVersionList, alreadyComputedPackage);
}
