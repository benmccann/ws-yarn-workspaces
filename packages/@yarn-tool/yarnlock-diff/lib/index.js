"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.YarnLockDiff = void 0;
const command_1 = require("@oclif/command");
const formatter_1 = require("./formatter");
const diff_service_1 = require("./diff-service");
const path_1 = require("path");
const fs_1 = require("fs");
class YarnLockDiff extends command_1.Command {
    constructor() {
        super(...arguments);
        this.checkIfExist = (filePathString) => {
            const filePath = path_1.resolve(process.cwd(), filePathString);
            return fs_1.existsSync(filePath);
        };
        this.reportNonExistantFiles = (filePathString) => {
            if (!this.checkIfExist(filePathString)) {
                this.error(`File does not exist "${filePathString}"`);
            }
            return filePathString;
        };
        this.checkIfLockFile = (filePathString) => {
            const fileExtension = path_1.extname(filePathString);
            return fileExtension === ".lock";
        };
        this.reportNonLockFile = (filePathString) => {
            if (!this.checkIfLockFile(filePathString)) {
                this.error(`File is not a lock file "${filePathString}"`);
            }
            return filePathString;
        };
    }
    async run() {
        const { flags } = this.parse(YarnLockDiff);
        const oldYarnLockPaths = flags.old;
        const newYarnLockPaths = flags.new;
        [...oldYarnLockPaths, ...newYarnLockPaths]
            .map(this.reportNonExistantFiles)
            .map(this.reportNonLockFile);
        const oldContents = oldYarnLockPaths.map(path => fs_1.readFileSync(path, "utf8"));
        const newContents = newYarnLockPaths.map(path => fs_1.readFileSync(path, "utf8"));
        diff_service_1.buildDiff(oldContents, newContents)
            .map(formatter_1.buildDiffTable)
            .map(this.log);
    }
}
exports.YarnLockDiff = YarnLockDiff;
YarnLockDiff.description = "Given one or more old yarn.lock files and one or more new yarn.lock files, compute the diff";
YarnLockDiff.flags = {
    version: command_1.flags.version({ char: "v" }),
    help: command_1.flags.help({ char: "h" }),
    old: command_1.flags.string({
        char: "o",
        description: "old yarn.lock file(s)",
        multiple: true,
        required: true,
    }),
    new: command_1.flags.string({
        char: "n",
        description: "new yarn.lock file(s)",
        multiple: true,
        required: true,
    }),
};
exports.default = YarnLockDiff;
//# sourceMappingURL=index.js.map