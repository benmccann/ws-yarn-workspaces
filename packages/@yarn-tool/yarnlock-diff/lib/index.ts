import { Command, flags } from "@oclif/command";

import { buildDiffTable } from './formatter';
import { buildDiff } from './diff-service';
import { extname, resolve } from 'path';
import { existsSync, readFileSync } from 'fs';

export class YarnLockDiff extends Command
{
	static description =
		"Given one or more old yarn.lock files and one or more new yarn.lock files, compute the diff";

	static flags = {
		version: flags.version({ char: "v" }),
		help: flags.help({ char: "h" }),
		old: flags.string({
			char: "o",
			description: "old yarn.lock file(s)",
			multiple: true,
			required: true,
		}),
		new: flags.string({
			char: "n",
			description: "new yarn.lock file(s)",
			multiple: true,
			required: true,
		}),
	};

	checkIfExist = (filePathString: string) =>
	{
		const filePath = resolve(process.cwd(), filePathString);
		return existsSync(filePath);
	};

	reportNonExistantFiles = (filePathString: string) =>
	{
		if (!this.checkIfExist(filePathString))
		{
			this.error(`File does not exist "${filePathString}"`);
		}
		return filePathString;
	};

	checkIfLockFile = (filePathString: string) =>
	{
		const fileExtension = extname(filePathString);
		return fileExtension === ".lock";
	};

	reportNonLockFile = (filePathString: string) =>
	{
		if (!this.checkIfLockFile(filePathString))
		{
			this.error(`File is not a lock file "${filePathString}"`);
		}
		return filePathString;
	};

	async run()
	{
		const { flags } = this.parse(YarnLockDiff);

		const oldYarnLockPaths = flags.old;
		const newYarnLockPaths = flags.new;

		[...oldYarnLockPaths, ...newYarnLockPaths]
			.map(this.reportNonExistantFiles)
			.map(this.reportNonLockFile);

		const oldContents = oldYarnLockPaths.map(path =>
			readFileSync(path, "utf8"),
		);
		const newContents = newYarnLockPaths.map(path =>
			readFileSync(path, "utf8"),
		);

		buildDiff(oldContents, newContents)
			.map(buildDiffTable)
			.map(this.log);
	}
}

export default YarnLockDiff;
