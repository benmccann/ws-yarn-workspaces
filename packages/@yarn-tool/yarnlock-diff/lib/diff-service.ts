import { diff, Diff } from "deep-diff";
import { buildComputedPackage } from './diff-service/buildComputedPackage';
import { ITSValueOrArray } from 'ts-type/lib/type/base';
import { IComputedPackage } from './diff-service/types';

export function buildDiff(
	oldYarnLockContent: ITSValueOrArray<Buffer | string>,
	newYarnLockContent: ITSValueOrArray<Buffer | string>,
)
{
	const oldPacakges = buildComputedPackage(oldYarnLockContent);

	const newPackages = buildComputedPackage(newYarnLockContent);

	return diff(oldPacakges, newPackages);
}
