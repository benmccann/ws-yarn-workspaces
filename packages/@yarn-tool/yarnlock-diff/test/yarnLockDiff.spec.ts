import { join } from 'path';
import FastGlob from '@bluelovers/fast-glob';
import { readFileSync } from 'fs';
import { buildDiff } from '../lib/diff-service';
import yarnLockDiff from '../index';
import { __TEST_YARNLOCK } from '../../../../__root_ws';

describe(`yarnLockDiff`, () =>
{
	const __res = join(__TEST_YARNLOCK, 'diff');

	const files = FastGlob.sync([
			'*.lock',
		], {
			cwd: join(__res, 'a'),
		})
	;

	files.forEach(file =>
	{

		test(file, () =>
		{
			const yarnlock_old = readFileSync(join(__res, 'a', file));
			const yarnlock_new = readFileSync(join(__res, 'b', file));

			let actual = yarnLockDiff(yarnlock_old, yarnlock_new, {
				stripAnsi: true,
			});

			expect(actual).toMatchSnapshot();

		});

	})
	;

})
