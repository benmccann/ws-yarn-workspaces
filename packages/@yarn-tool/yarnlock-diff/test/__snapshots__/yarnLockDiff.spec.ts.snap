// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yarnLockDiff v1.iconv.lock 1`] = `
" package name                                        old version(s)           new version(s) 
 @babel/core                                            7.12.16            →      7.14.3     
 @babel/generator                                       7.12.15            →      7.14.3     
 @babel/helper-function-name                            7.12.13            →      7.14.2     
 @babel/helper-member-expression-to-functions           7.12.16            →     7.13.12     
 @babel/helper-module-imports                           7.12.13            →     7.13.12     
 @babel/helper-module-transforms                        7.12.13            →      7.14.2     
 @babel/helper-plugin-utils                             7.12.13            →      7.13.0     
 @babel/helper-replace-supers                           7.12.13            →      7.14.4     
 @babel/helper-simple-access                            7.12.13            →     7.13.12     
 @babel/helper-validator-identifier                     7.12.11            →      7.14.0     
 @babel/helpers                                         7.12.13            →      7.14.0     
 @babel/highlight                                       7.12.13            →      7.14.0     
 @babel/parser                                          7.12.16            →      7.14.4     
 @babel/traverse                                        7.12.13            →      7.14.2     
 @babel/types                                           7.12.13            →      7.14.4     
 @cnakazawa/watch                                        1.0.4             →        -        
 @istanbuljs/schema                                      0.1.2             →      0.1.3      
 @jest/console                                           26.6.2            →      27.0.2     
 @jest/core                                              26.6.3            →      27.0.4     
 @jest/environment                                       26.6.2            →      27.0.3     
 @jest/fake-timers                                       26.6.2            →      27.0.3     
 @jest/globals                                           26.6.2            →      27.0.3     
 @jest/reporters                                         26.6.2            →      27.0.4     
 @jest/source-map                                        26.6.2            →      27.0.1     
 @jest/test-result                                       26.6.2            →      27.0.2     
 @jest/test-sequencer                                    26.6.3            →      27.0.4     
 @jest/transform                                         26.6.2            →      27.0.2     
 @jest/types                                             [...]             →  [..., 27.0.2]  
 @sinonjs/commons                                        1.8.2             →      1.8.3      
 @sinonjs/fake-timers                                    6.0.1             →      7.1.2      
 @types/babel__core                                      7.1.12            →      7.1.14     
 @types/babel__traverse                                  7.11.0            →      7.11.1     
 @types/bluebird                                         3.5.33            →      3.5.35     
 @types/fs-extra                                         9.0.7             →      9.0.11     
 @types/istanbul-reports                                 3.0.0             →      3.0.1      
 @types/jest                                            26.0.20            →     26.0.23     
 @types/lodash                                          4.14.168           →     4.14.170    
 @types/luxon                                            1.25.1            →      1.26.5     
 @types/node                                            14.14.26           →     15.12.1     
 @types/normalize-package-data                           2.4.0             →        -        
 @types/prettier                                         2.2.0             →      2.2.3      
 @types/yargs                                            [...]             →  [..., 16.0.3]  
 acorn                                                   [...]             →   [..., 8.3.0]  
 ajv                                                     6.12.6            →        -        
 ansi-escapes                                            4.3.1             →      4.3.2      
 ansi-styles                                             [...]             →   [..., 5.2.0]  
 anymatch                                                2.0.0             →      3.1.2      
 arr-diff                                                4.0.0             →        -        
 arr-flatten                                             1.1.0             →        -        
 arr-union                                               3.1.0             →        -        
 array-unique                                            0.3.2             →        -        
 asn1                                                    0.2.4             →        -        
 assert-plus                                             1.0.0             →        -        
 assign-symbols                                          1.0.0             →        -        
 at-least-node                                           1.0.0             →        -        
 atob                                                    2.1.2             →        -        
 aws-sign2                                               0.7.0             →        -        
 aws4                                                    1.11.0            →        -        
 babel-jest                                              26.6.3            →      27.0.2     
 babel-plugin-jest-hoist                                 26.6.2            →      27.0.1     
 babel-preset-jest                                       26.6.2            →      27.0.1     
 balanced-match                                          1.0.0             →      1.0.2      
 base                                                    0.11.2            →        -        
 bcrypt-pbkdf                                            1.0.2             →        -        
 braces                                                  2.3.2             →      3.0.2      
 cache-base                                              1.0.1             →        -        
 capture-exit                                            2.0.0             →        -        
 caseless                                                0.12.0            →        -        
 chalk                                                   4.1.0             →      4.1.1      
 ci-info                                                 2.0.0             →      3.2.0      
 cjs-module-lexer                                        0.6.0             →      1.2.1      
 class-utils                                             0.3.6             →        -        
 cliui                                                   6.0.0             →      7.0.4      
 collection-visit                                        1.0.0             →        -        
 component-emitter                                       1.3.0             →        -        
 copy-descriptor                                         0.1.1             →        -        
 core-util-is                                            1.0.2             →        -        
 cross-spawn                                             6.0.5             →      7.0.3      
 dashdash                                                1.14.1            →        -        
 debug                                                   2.6.9             →      4.3.1      
 decamelize                                              1.2.0             →        -        
 decode-uri-component                                    0.2.0             →        -        
 define-property                                  0.2.5, 1.0.0, 2.0.2      →        -        
 diff-sequences                                          [...]             →  [..., 27.0.1]  
 ecc-jsbn                                                0.1.2             →        -        
 emittery                                                0.7.2             →      0.8.1      
 end-of-stream                                           1.4.4             →        -        
 error-ex                                                1.3.2             →        -        
 escodegen                                               1.14.3            →      2.0.0      
 estraverse                                              4.3.0             →      5.2.0      
 exec-sh                                                 0.3.4             →        -        
 execa                                                   1.0.0             →      5.1.1      
 expand-brackets                                         2.1.4             →        -        
 expect                                                  26.6.2            →      27.0.2     
 extend-shallow                                       2.0.1, 3.0.2         →        -        
 extend                                                  3.0.2             →        -        
 extglob                                                 2.0.4             →        -        
 extsprintf                                           1.3.0, 1.4.0         →        -        
 fast-deep-equal                                         3.1.3             →        -        
 fill-range                                              4.0.0             →      7.0.1      
 for-in                                                  1.0.2             →        -        
 forever-agent                                           0.6.1             →        -        
 form-data                                               2.3.3             →      3.0.1      
 fragment-cache                                          0.2.1             →        -        
 fs-extra                                                9.1.0             →      10.0.0     
 get-stream                                              4.1.0             →      6.0.1      
 get-value                                               2.0.6             →        -        
 getpass                                                 0.1.7             →        -        
 glob                                                    7.1.6             →      7.1.7      
 growly                                                  1.3.0             →        -        
 har-schema                                              2.0.0             →        -        
 har-validator                                           5.1.5             →        -        
 has-value                                            0.3.1, 1.0.0         →        -        
 has-values                                           0.1.4, 1.0.0         →        -        
 hosted-git-info                                         2.8.8             →        -        
 http-signature                                          1.2.0             →        -        
 human-signals                                           1.1.1             →      2.1.0      
 iconv-lite                                              0.6.2             →      0.6.3      
 ip-regex                                                2.1.0             →        -        
 is-accessor-descriptor                               0.1.6, 1.0.0         →        -        
 is-arrayish                                             0.2.1             →        -        
 is-buffer                                               1.1.6             →        -        
 is-ci                                                   2.0.0             →      3.0.0      
 is-core-module                                          2.2.0             →      2.4.0      
 is-data-descriptor                                   0.1.4, 1.0.0         →        -        
 is-descriptor                                        0.1.6, 1.0.2         →        -        
 is-docker                                               2.1.1             →        -        
 is-extendable                                        0.1.1, 1.0.1         →        -        
 is-number                                               3.0.0             →      7.0.0      
 is-plain-object                                         2.0.4             →        -        
 is-potential-custom-element-name                        1.0.0             →      1.0.1      
 is-stream                                               1.1.0             →      2.0.0      
 is-windows                                              1.0.2             →        -        
 is-wsl                                                  2.2.0             →        -        
 isarray                                                 1.0.0             →        -        
 isobject                                             2.1.0, 3.0.1         →        -        
 isstream                                                0.1.2             →        -        
 jest-changed-files                                      26.6.2            →      27.0.2     
 jest-cli                                                26.6.3            →      27.0.4     
 jest-config                                             26.6.3            →      27.0.4     
 jest-diff                                               [...]             →  [..., 27.0.2]  
 jest-docblock                                           26.0.0            →      27.0.1     
 jest-each                                               26.6.2            →      27.0.2     
 jest-environment-jsdom                                  26.6.2            →      27.0.3     
 jest-environment-node                                   26.6.2            →      27.0.3     
 jest-get-type                                           [...]             →  [..., 27.0.1]  
 jest-haste-map                                          26.6.2            →      27.0.2     
 jest-jasmine2                                           26.6.3            →      27.0.4     
 jest-leak-detector                                      26.6.2            →      27.0.2     
 jest-matcher-utils                                      26.6.2            →      27.0.2     
 jest-message-util                                       26.6.2            →      27.0.2     
 jest-mock                                               26.6.2            →      27.0.3     
 jest-regex-util                                         26.0.0            →      27.0.1     
 jest-resolve-dependencies                               26.6.3            →      27.0.4     
 jest-resolve                                            26.6.2            →      27.0.4     
 jest-runner                                             26.6.3            →      27.0.4     
 jest-runtime                                            26.6.3            →      27.0.4     
 jest-serializer                                         26.6.2            →      27.0.1     
 jest-snapshot                                           26.6.2            →      27.0.4     
 jest-util                                               26.6.2            →      27.0.2     
 jest-validate                                           26.6.2            →      27.0.2     
 jest-watcher                                            26.6.2            →      27.0.2     
 jest-worker                                             26.6.2            →      27.0.2     
 jest                                                    26.6.3            →      27.0.4     
 jsbn                                                    0.1.1             →        -        
 jschardet                                               2.2.1             →      3.0.0      
 jsdom                                                   16.4.0            →      16.6.0     
 json-parse-even-better-errors                           2.3.1             →        -        
 json-schema-traverse                                    0.4.1             →        -        
 json-schema                                             0.2.3             →        -        
 json-stringify-safe                                     5.0.1             →        -        
 jsprim                                                  1.4.1             →        -        
 kind-of                                       3.2.2, 4.0.0, 5.1.0, 6.0.3  →        -        
 lines-and-columns                                       1.1.6             →        -        
 lodash.sortby                                           4.7.0             →        -        
 lodash                                                 4.17.20            →     4.17.21     
 luxon                                                   1.25.0            →      1.27.0     
 map-cache                                               0.2.2             →        -        
 map-visit                                               1.0.0             →        -        
 micromatch                                              3.1.10            →      4.0.4      
 mime-db                                                 1.45.0            →      1.48.0     
 mime-types                                              2.1.28            →      2.1.31     
 mixin-deep                                              1.3.2             →        -        
 ms                                                      2.0.0             →      2.1.2      
 nanomatch                                               1.2.13            →        -        
 nice-try                                                1.0.5             →        -        
 node-notifier                                           8.0.1             →        -        
 normalize-package-data                                  2.5.0             →        -        
 normalize-path                                          2.1.1             →      3.0.0      
 npm-run-path                                            2.0.2             →      4.0.1      
 oauth-sign                                              0.9.0             →        -        
 object-copy                                             0.1.0             →        -        
 object-visit                                            1.0.1             →        -        
 object.pick                                             1.3.0             →        -        
 p-finally                                               1.0.0             →        -        
 parse-json                                              5.2.0             →        -        
 parse5                                                  5.1.1             →      6.0.1      
 pascalcase                                              0.1.1             →        -        
 path-key                                                2.0.1             →      3.1.1      
 path-parse                                              1.0.6             →      1.0.7      
 performance-now                                         2.1.0             →        -        
 picomatch                                               2.2.2             →      2.3.0      
 posix-character-classes                                 0.1.1             →        -        
 pretty-format                                           [...]             →  [..., 27.0.2]  
 prompts                                                 2.4.0             →      2.4.1      
 pump                                                    3.0.0             →        -        
 qs                                                      6.5.2             →        -        
 react-is                                                17.0.1            →      17.0.2     
 read-pkg-up                                             7.0.1             →        -        
 read-pkg                                                5.2.0             →        -        
 regex-not                                               1.0.2             →        -        
 remove-trailing-separator                               1.1.0             →        -        
 repeat-element                                          1.1.3             →        -        
 repeat-string                                           1.6.1             →        -        
 request-promise-core                                    1.1.4             →        -        
 request-promise-native                                  1.0.9             →        -        
 request                                                 2.88.2            →        -        
 require-main-filename                                   2.0.0             →        -        
 resolve-url                                             0.2.1             →        -        
 ret                                                     0.1.15            →        -        
 rsvp                                                    4.8.5             →        -        
 safe-buffer                                          [..., 5.2.1]         →      [...]      
 safe-regex                                              1.1.0             →        -        
 sane                                                    4.1.0             →        -        
 semver                                                  5.7.1             →      6.3.0      
 set-blocking                                            2.0.0             →        -        
 set-value                                               2.0.1             →        -        
 shebang-command                                         1.2.0             →      2.0.0      
 shebang-regex                                           1.0.0             →      3.0.0      
 shellwords                                              0.1.1             →        -        
 snapdragon-node                                         2.1.1             →        -        
 snapdragon-util                                         3.0.1             →        -        
 snapdragon                                              0.8.2             →        -        
 source-map-resolve                                      0.5.3             →        -        
 source-map-url                                          0.4.1             →        -        
 spdx-correct                                            3.1.1             →        -        
 spdx-exceptions                                         2.3.0             →        -        
 spdx-expression-parse                                   3.0.1             →        -        
 spdx-license-ids                                        3.0.7             →        -        
 split-string                                            3.1.0             →        -        
 sshpk                                                   1.16.1            →        -        
 static-extend                                           0.1.2             →        -        
 stealthy-require                                        1.1.1             →        -        
 string-length                                           4.0.1             →      4.0.2      
 string-width                                            4.2.0             →      4.2.2      
 strip-eof                                               1.0.0             →        -        
 supports-color                                          [...]             →   [..., 8.1.1]  
 supports-hyperlinks                                     2.1.0             →      2.2.0      
 throat                                                  5.0.0             →      6.0.1      
 to-object-path                                          0.3.0             →        -        
 to-regex-range                                          2.1.1             →      5.0.1      
 to-regex                                                3.0.2             →        -        
 tough-cookie                                            2.5.0             →      4.0.0      
 tr46                                                    2.0.2             →      2.1.0      
 ts-jest                                                 26.5.1            →      27.0.2     
 tunnel-agent                                            0.6.0             →        -        
 tweetnacl                                               0.14.5            →        -        
 type-fest                                               0.6.0             →      0.21.3     
 union-value                                             1.0.1             →        -        
 universalify                                            2.0.0             →      0.1.2      
 unset-value                                             1.0.0             →        -        
 uri-js                                                  4.4.1             →        -        
 urix                                                    0.1.0             →        -        
 use                                                     3.1.1             →        -        
 uuid                                                 3.4.0, 8.3.2         →        -        
 v8-to-istanbul                                          7.1.0             →      7.1.2      
 validate-npm-package-license                            3.0.4             →        -        
 verror                                                  1.10.0            →        -        
 whatwg-url                                              8.4.0             →      8.5.0      
 which-module                                            2.0.0             →        -        
 which                                                   1.3.1             →      2.0.2      
 wrap-ansi                                               6.2.0             →      7.0.0      
 ws                                                      7.4.3             →      7.4.6      
 y18n                                                    4.0.1             →      5.0.8      
 yargs-parser                                            18.1.3            →      20.2.7     
 yargs                                                   15.4.1            →      16.2.0     
 @babel/compat-data                                        -               →      7.14.4     
 @babel/helper-compilation-targets                         -               →      7.14.4     
 @babel/helper-validator-option                            -               →     7.12.17     
 @babel/plugin-syntax-typescript                           -               →     7.12.13     
 @tootallnate/once                                         -               →      1.1.2      
 agent-base                                                -               →      6.0.2      
 browserslist                                              -               →      4.16.6     
 caniuse-lite                                              -               →   1.0.30001234  
 colorette                                                 -               →      1.2.2      
 dedent                                                    -               →      0.7.0      
 electron-to-chromium                                      -               →     1.3.749     
 escalade                                                  -               →      3.1.1      
 http-proxy-agent                                          -               →      4.0.1      
 https-proxy-agent                                         -               →      5.0.0      
 jest-circus                                               -               →      27.0.4     
 node-releases                                             -               →      1.1.72     "
`;

exports[`yarnLockDiff v2.lock 1`] = `
" package name          old version(s)     new version(s) 
 @bluelovers/tsconfig      1.0.19      →      1.0.20     "
`;

exports[`yarnLockDiff yarn.lock 1`] = `
" package name       old version(s)     new version(s) 
 @babel/code-frame      7.0.2       →      7.0.0      
 @babel/generator       6.3.7       →      6.3.4      
 @babel/types           7.3.4       →        -        
 @babel/parser            -         →      7.3.4      "
`;
